inc/saipoe.h

```diff
@@ -0,0 +1,524 @@
+     * @default SAI_POE_DEVICE_LIMIT_MODE_TYPE_CLASS_LIMIT
+     */
+    SAI_POE_DEVICE_ATTR_POWER_LIMIT_MODE,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_POE_DEVICE_ATTR_END,
+
+    /** Custom range base value */
+    SAI_POE_DEVICE_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_POE_DEVICE_ATTR_CUSTOM_RANGE_END
+
+} sai_poe_device_attr_t;
+
+/**
+ * @brief poe PSE attributes
+ */
+typedef enum _sai_poe_pse_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_POE_PSE_ATTR_START,
+
+    /**
+     * @brief PoE device ID
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_POE_PSE_ATTR_DEVICE_ID = SAI_POE_PSE_ATTR_START,
+
+    /**
+     * @brief PoE PSE ID (index)
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE ONLY
+     */
+    SAI_POE_PSE_ATTR_ID,
+    
+    /**
+     * @brief poe pse PSE software version
+     *
+     * @type char
+     * @flags READ_ONLY
+     */
+    SAI_POE_PSE_ATTR_PSE_SOFTWARE_VERSION,
+
+    /**
+     * @brief poe pse PSE hardware version
+     *
+     * @type char
+     * @flags READ_ONLY
+     */
+    SAI_POE_PSE_ATTR_PSE_HARDWARE_VERSION,
+
+        /**
+     * @brief temperature (in celsius) of the PSE
+     *
+     * @type int16_t
+     * @flags READ_ONLY
+     */
+    SAI_POE_PSE_ATTR_TEMPERATURE,
+
+    /**
+     * @brief status of the PSE
+     *
+     * @type sai_poe_device_pse_status_t
+     * @flags READ_ONLY
+     */
+    SAI_POE_PSE_ATTR_PSE_STATUS,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_POE_PSE_ATTR_END,
+
+    /** Custom range base value */
+    SAI_POE_PSE_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_POE_PSE_ATTR_CUSTOM_RANGE_END
+
+} sai_poe_pse_attr_t;
+
+
+/**
+ * @brief poe port attributes
+ */
+typedef enum _sai_poe_port_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_POE_PORT_ATTR_START,
+
+    /**
+     * @brief PoE device ID
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_POE_ATTR_DEVICE_ID = SAI_POE_PORT_ATTR_START,
+
+    /**
+     * @brief PoE port front panel ID
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE ONLY
+     */
+    SAI_POE_PORT_ATTR_FRONT_PANEL_ID,
+
+    /**
+     * @brief PoE port standard (af/at/bt...)
+     *
+     * @type sai_poe_port_standard_t
+     * @flags READ_ONLY
+     * @default SAI_POE_PORT_STANDARD_TYPE_AT
+     */
+    SAI_POE_PORT_ATTR_STANDARD,
+
+    /**
+     * @brief PoE port enabled/disabled state, as set by the user
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_POE_PORT_ATTR_ADMIN_ENABLED_STATE,
+
+    /**
+     * @brief PoE port power limit mode
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default maximum value that the port can provide
```

Comment:
default can be left as a special value, say 0, to indicate that no max limit is enforced.

Reply:
Done, thanks.